src/csharp/Grpc.Tools/GeneratorServices.cs

```diff
@@ -109,6 +108,26 @@ string LowerUnderscoreToUpperCamel(string str)
             }
             return result.ToString();
         }
+
+        // This is how the protoc codegen constructs its output filename.
+        // See protobuf/compiler/csharp/csharp_helpers.cc:356.
+        // Note that protoc explicitly discards non-ASCII letters.
+        string LowerUnderscoreToUpperCamelProtocWay(string str)
+        {
+            var result = new StringBuilder(str.Length, str.Length);
+            bool cap = true;
+            foreach (char c in str)
+            {
+                char upperC = char.ToUpperInvariant(c);
+                bool isAsciiLetter = 'A' <= upperC && upperC <= 'Z';
+                if (isAsciiLetter || ('0' <= c && c <= '9'))
+                {
+                    result.Append(cap ? upperC : c);
+                }
+                cap = !isAsciiLetter;
```

Comment:
I believe there's still a subtle problem - after a digit, protoc will still capitalize the next letter:
I tested and `one123two.cs` generates `One123Two.cs` and `One123twoGrpc.cs`.

Reply:
Uhhhh, thanks! I'll take care of this.

Reply:
This passess, I added a test and checked in real build, too. `cap = !isAsciiLetter;` will be true after a digit, as well as after any (non-emitted) other character. I'll push the test case tho, so it exists.

Reply:
Done, aa40424.

Reply:
ah, thanks for clarification!